pkg/sql/mvcc_backfiller.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sql
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+	"github.com/cockroachdb/cockroach/pkg/sql/rowenc"
+	"github.com/cockroachdb/errors"
+)
+
+// Merge merges the entries in the destination index to the source index. The
+// destination index is always assumed to be using the delete-preserving
+// encoding. Key conflicts are resolved by always preferring the value in the
+// destination index, or by preferring the delete if applicable.
+func (sc *SchemaChanger) Merge(
+	ctx context.Context,
+	codec keys.SQLCodec,
+	table catalog.TableDescriptor,
+	source catalog.Index,
+	destination catalog.Index,
+) error {
+	// TODO(rhu): Do we need to get the timestamp and use a fix timestamp?
+	mergeTimestamp := sc.clock.Now()
+
+	sourceSpan := table.IndexSpan(codec, source.GetID())
+	destSpan := table.IndexSpan(codec, destination.GetID())
+
+	const pageSize = 1000
+
+	return sc.db.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
+		if err := txn.SetFixedTimestamp(ctx, mergeTimestamp); err != nil {
+			return err
+		}
+
+		return txn.Iterate(ctx, sourceSpan.Key, sourceSpan.EndKey, pageSize, func(kvs []kv.KeyValue) error {
```

Comment:
Does this iterate function mean we are doing everything in a single transaction? Is that what we want?

Reply:
Changed to use 1 transaction per `pageSize`.